File: java/src/LC_14_LongestCommonPrefix.java
// https://leetcode.com/problems/longest-common-prefix/

public class LC_14_LongestCommonPrefix {
    public String longestCommonPrefix(String[] strs) {
        int i;
        for (i = 0; i < strs[0].length(); i++) {
            char ch = strs[0].charAt(i);
            boolean isMisMatchFound = false;
            for (int j = 1; j < strs.length; j++) {
                String str = strs[j];
                if (i >= str.length() || str.charAt(i) != ch) {
                    isMisMatchFound = true;
                    break;
                }
            }
            if (isMisMatchFound)
                break;
        }
        return strs[0].substring(0, i);
    }
}

File: python/LC_14_LongestCommonPrefix.py
# https://leetcode.com/problems/longest-common-prefix/
from typing import List


class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        prefix = strs[0]
        j = 0
        while j < len(prefix):
            is_mismatch_found = False
            for i in range(1, len(strs)):
                if j >= len(strs[i]) or strs[i][j] != prefix[j]:
                    is_mismatch_found = True
                    break
            if is_mismatch_found:
                break
            j += 1
        return prefix[0:j]


sol = Solution()
print(sol.longestCommonPrefix(["flower", "flow", "flight"]))
print(sol.longestCommonPrefix(strs=["dog", "racecar", "car"]))
